barretenberg/cpp/src/barretenberg/vm/avm/trace/gadgets/range_check.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "barretenberg/vm/avm/generated/relations/range_check.hpp"
+#include "barretenberg/vm/avm/trace/common.hpp"
+#include <cstdint>
+
+enum class EventEmitter {
+    ALU,
+    MEMORY,
+};
+
+namespace bb::avm_trace {
+// template <class TraceUtilisingRange>
+class AvmRangeCheckBuilder {
+  public:
+    struct RangeCheckEvent {
+        uint32_t clk;
+        FF value;
```

Comment:
I'm recently thinking that we could try to make the events as small as possible. E.g., here you can use a uint128_t?